### DISCUSSION TITLE
An error comes out when run  the create_forcing.exe

### DISCUSSION THREAD

**Discussion by Liyuyu3** [NCAR/hrldas#143] [Q&A] [answered]
Hello, I followed /NCAR/hrldas/hrldas/README.ERA5 to run Noah-mp and this error occurred while making the forcing data.
![770a537a71945cdcf150a0e4c0eaf66](https://github.com/NCAR/hrldas/assets/154398859/80ba1f91-6a50-4044-8db3-9efc8f01ecba)

**Comment by cenlinhe** [comment]
Based on your error message, this is because the terrain data input file is not found. Please double check the terrain data file path in your namelist to make sure it is correct based on the directory on your local machine.

**Reply by Liyuyu3** [reply]
I check the file(geo_1279l4_0.1x0.1.grib2_v4_unpack.nc) path and make sure it is correct, but still come out this error.
![Uploading f3cc2c1196444cc7211fbdc0258975f.jpg…]()

**Reply by Liyuyu3** [reply]
Thanks for your reply! I saw the same question before and it said "The create_forcing.exe  requires to read the name "TERRAIN" from the Vtable in the namelist.input and then it reads the geopotential z from the Zfile_template file in the namelist. The create_forcing.exe then convert the geopotential to the height. For both Zfile_template and LANDSfile_template, they require NetCDF files rather than grib files for generating ERA5 forcing."(based on @tslin2   I follow this to set my namelist.input, but it still doesn't work.

**Reply by tslin2** [reply]
Hi, 

Looks like your filename is fine, do you have the read permission of this file geo_1279l4_0.1x0.1.grib2_v4_unpack.nc

this command can check the permission
ls -l http://geo_1279l4_0.1x0.1.grib2_v4_unpack.nc

or it could be this file is corrupt, so try downloading again.

**Reply by cenlinhe** [reply]
hmm... I just tested on my end and everything works fine. You may try to re-download the geo_1279l4_0.1x0.1.grib2_v4_unpack.nc and lsm_1279l4_0.1x0.1.grb_v4_unpack.nc. If it is still not working, you can download those two files here from our community Noah-MP google drive: https://drive.google.com/drive/folders/1ZghXJ2NApdFPdkOC-DKpSj_p7AT_yQ_0

**Reply by Liyuyu3** [reply]
I have also considered it to be an issue with file permissions or the file itself is broken. However, the file permission is -rw-rw-r-- which means I am able to read it, and I also try to download this file on https://drive.google.com/drive/folders/1ZghXJ2NApdFPdkOC-DKpSj_p7AT_yQ_0, the error is still not resolved.

**Reply by cenlinhe** [reply]
This is weird. Are you able to use ncview to successfully open and view those netcdf files?

**Reply by tslin2** [reply]
You may check the NetCDF version; is it  version 4?

**Reply by Liyuyu3** [reply]
These files can be viewed normally using the command "nudmp -h", and I checked the version by entering the command "nc-config --version" ,it is netCDF 4.8.1. I really don't understand why something went wrong.

**Reply by tslin2** [reply]
is it the NetCDF Fortran library?!

**Reply by tslin2** [reply]
maybe can also try changing the file from classic netCDF-4 model to netCDF-4 to see if works

ncdump -k geo_1279l4_0.1x0.1.grib2_v4_unpack.nc

nccopy -k 3 geo_1279l4_0.1x0.1.grib2_v4_unpack.nc geo_1279l4_0.1x0.1.grib2_v4_unpack1.nc

**Reply by Liyuyu3** [reply]
OK. I will try it. My NetCDF fortran is 4.5.3 and c is 4.7.4.

**Reply by Liyuyu3** [reply]
It doesn't work, this error still occurs. I really don't understand what went wrong.

**Reply by tslin2** [reply, marked as answer]
The issue is resolved in the email exchange by changing the nc file format from netCDF-4 classic model to classic file format.

nccopy -k classic geo_1279l4_0.1x0.1.grib2_v4_unpack.nc geo_1279l4_0.1x0.1.grib2_v4_unpack1.nc

and revise the namelist to geo_1279l4_0.1x0.1.grib2_v4_unpack1.nc

Zfile_template     = "./geo_1279l4_0.1x0.1.grib2_v4_unpack1.nc

**Reply by cenlinhe** [reply]
Great. Is this because the specific netcdf version on the user's computer? We do not have this issue on our machine.

**Reply by tslin2** [reply]
I am not sure, but the error from nc_open is -128

#define NC_ENOTBUILT     (-128)    // Attempt to use feature that was not turned on when netCDF was built.

https://docs.unidata.ucar.edu/netcdf/documentation/4.8.0/nc-error-codes.html#NetCDF-4%20Error%20Codes
